test/e2e/internal/notation/host.go

```diff
@@ -0,0 +1,109 @@
+package notation
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/notaryproject/notation/test/e2e/internal/utils"
+)
+
+// Host creates a virtualized notation testing host by modify
+// the "XDG_CONFIG_HOME" environment variable of the Executor.
+//
+// options is the required testing environment options
+// fn is the callback function containing the testing logic.
+func Host(options []utils.HostOption, fn func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost)) {
```

Comment:
Just like https://pkg.go.dev/path/filepath#WalkFunc, can we define a function so that the meaning of `fn` is more clear? Eg
```
type CoreTestFunc func(vhost *utils.VirtualHost, artifact *Artifact, notation *utils.ExecOpts)
func Host(options []utils.HostOption, CoreTestFunc) {
// ...
}
```
Also I want to reorder the parameter. 
`vhost` is your environment, so put it in the first place.

Reply:
Updated.